src/components/WarrantyForm/useWarrantyForm.ts

```diff
@@ -0,0 +1,47 @@
+import { useState } from 'react';
+import { FieldValues } from 'react-hook-form';
+
+const useWarrantyForm = (captchaToken: string | null) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [message, setMessage] = useState('');
+
+  const onSubmit = async (data: FieldValues) => {
```

Comment:
Same notes here as in PR 1. If possible move async functions out of components, and handle state in the caller

Reply:
Custom hooks can handle state in the hook definition hence it's moved here.